src/server.js

```diff
@@ -27,22 +27,87 @@ app.get('/', (req, res) => {
 });
 
 app.post('/api/chat', checkIsAuthenticated, async (req, res) => {
-  const { message } = req.body;
+  const { message, user_id, conversation_id } = req.body;
 
   if (!message) {
     return res.status(400).json({ error: "Message is required." });
   }
 
+  if (!user_id) {
+    return res.status(400).json({ error: "User ID is required." });
+  }
+
   try {
     const openaiResponse = await openaiRequest(message);
-    res.json({ response: openaiResponse });
+
+    let associatedConversationID;
+    let continuedConversation;
+    //if no conversation ID present, add a new conversation and use its ID - otherwise use the ID given
+    if (!conversation_id) {
+      continuedConversation = false;
+      const openaiSummary = await openaiRequest("Summarize the specific topic of the question that was asked in 5 words or less.");
```

Comment:
Here you could call my recent PR from vicente/add-tittlle-generator to get the title